src/pathfinding_service/model/token_network.py

```diff
@@ -79,11 +79,17 @@ def __init__(
         self.value = value
         self.address_to_reachability = address_to_reachability
         self.fees: List[FeeAmount] = []
-        self._calculate_fees()
+        self._is_valid = self._check_validity()
 
-        log.debug("Creating Path object", nodes=nodes, is_valid=self.is_valid, fees=self.fees)
+        log.debug("Creating Path object", nodes=nodes, is_valid=self._is_valid, fees=self.fees)
 
-    def _calculate_fees(self) -> None:
+    def _calculate_fees(self) -> bool:
```

Comment:
If we made this
```suggestion
    def _calculate_fees(self) -> Optional[FeeAmount]:
```
would not modify any state, anymore.